packages/salesforcedx-vscode-core/src/commands/forceSourceDeploySourcePath.ts

```diff
@@ -71,6 +72,20 @@ export class MultipleSourcePathsGatherer implements ParametersGatherer<string> {
   }
 }
 
+export class LibraryPathsGatherer implements ParametersGatherer<string[]> {
+  private uris: vscode.Uri[];
+  public constructor(uris: vscode.Uri[]) {
+    this.uris = uris;
+  }
+  public async gather(): Promise<ContinueResponse<string[]>> {
+    const sourcePaths = this.uris.map(uri => uri.fsPath);
+    return {
+      type: 'CONTINUE',
+      data: sourcePaths
+    };
+  }
+}
```

Comment:
I haven't done it in this PR but we should explore having a single gatherer no matter the number of files being deployed